extide/gradle/src/org/netbeans/modules/gradle/ProjectTrust.java

```diff
@@ -65,14 +72,31 @@
         projectTrust = prefs.node(NODE_PROJECT);
         key = new SecretKeySpec(salt, HMAC_SHA256);
     }
-
+    
     /**
      * Returns true if the specified project is trusted.
      *
      * @param project of the trust check.
      * @return true if the given project is trusted.
      */
     public boolean isTrusted(Project project) {
+        synchronized (this) {
+            if (temporaryTrustedIds.contains(getPathId(project))) {
+                return true;
+            }
+        }
+        return isTrustedPermanetly(project);
+    }
+
+    /**
+     * Returns true if the specified project is trusted <b>permanently</b>.
+     * You should be probably using {@link #isTrusted} to avoid duplicate questions
+     * during one IDE run.
+     *
+     * @param project of the trust check.
+     * @return true if the given project is trusted.
+     */
+    public boolean isTrustedPermanetly(Project project) {
```

Comment:
Just noticed this method name has a typo (it was highlighted while using `git show` :smile: ).